dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/dxf2/metadata/objectbundle/validation/ReferencesCheckTest.java

```diff
@@ -0,0 +1,120 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.dxf2.metadata.objectbundle.validation;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.hisp.dhis.DhisSpringTest;
+import org.hisp.dhis.common.IdentifiableObject;
+import org.hisp.dhis.dxf2.metadata.objectbundle.ObjectBundle;
+import org.hisp.dhis.dxf2.metadata.objectbundle.ObjectBundleHooks;
+import org.hisp.dhis.dxf2.metadata.objectbundle.ObjectBundleParams;
+import org.hisp.dhis.feedback.ErrorCode;
+import org.hisp.dhis.feedback.ErrorReport;
+import org.hisp.dhis.feedback.ObjectReport;
+import org.hisp.dhis.feedback.TypeReport;
+import org.hisp.dhis.importexport.ImportStrategy;
+import org.hisp.dhis.indicator.Indicator;
+import org.hisp.dhis.indicator.IndicatorType;
+import org.hisp.dhis.preheat.Preheat;
+import org.hisp.dhis.schema.SchemaService;
+import org.hisp.dhis.schema.validation.SchemaValidator;
+import org.hisp.dhis.security.acl.AclService;
+import org.hisp.dhis.user.UserService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.google.common.collect.Lists;
+
+public class ReferencesCheckTest extends DhisSpringTest
+{
+    @Autowired
+    private SchemaValidator schemaValidator;
+
+    @Autowired
+    private SchemaService schemaService;
+
+    @Autowired
+    private AclService aclService;
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private ObjectBundleHooks objectBundleHooks;
+
+    private ValidationContext validationContext;
+
+    @BeforeEach
+    public void beforeEach()
+    {
+        validationContext = new ValidationContext( this.objectBundleHooks, this.schemaValidator, this.aclService,
+            this.userService,
+            this.schemaService );
+    }
+
+    @Test
+    void testCheckReferenceErrorReport()
+    {
+        ObjectBundle bundle = createObjectBundle();
+        Indicator indicator = new Indicator();
+        indicator.setUid( "indicatorUid" );
+        indicator.setName( "indicatorA" );
+        IndicatorType indicatorType = new IndicatorType();
+        indicator.setName( "indicatorTypeA" );
+        indicator.setIndicatorType( indicatorType );
+
+        ReferencesCheck referencesCheck = new ReferencesCheck();
+        TypeReport report = referencesCheck.check( bundle, Indicator.class, Collections.emptyList(),
+            Lists.newArrayList( indicator ), ImportStrategy.CREATE, validationContext );
```

Comment:
You can now use `List.of` instead of `Lists.newArrayList` as long as you know for sure that the elements are all non-`null`.